### PR TITLE
Refactor queue module

### PR DESCRIPTION
**Pull request by Androbin**
Towards #371

### CODE REVIEW

**Comment by sushain97** [comment]
The diff here is basically useless lol. Could you either add some tests or post a transcript of the module still working as expected?

**Comment by Androbin** [comment]
```
<androbin> .queue display foo
<phenny> androbin: No queues found.
<androbin> .queue new todo
<phenny> androbin: Empty queue androbin:todo created.
<androbin> .queue new doto lorem, ipsum
<phenny> androbin: Queue androbin:doto with items lorem, ipsum created.
<androbin> .queue display todo
<phenny> [androbin:todo] - <empty>
<androbin> .queue display doto
<phenny> [androbin:doto] - lorem, ipsum
<androbin> .queue todo add dolor, sit, amet
<phenny> [androbin:todo] - dolor, sit, amet
<androbin> .queue doto swap ipsum, lorem
<phenny> [androbin:doto] - ipsum, lorem
<androbin> .queue todo move dolor, amet
<phenny> [androbin:todo] - sit, amet, dolor
<androbin> .queue doto swap 0, 1
<phenny> [androbin:doto] - lorem, ipsum
<androbin> .queue todo move 2, 1
<phenny> [androbin:todo] - sit, dolor, amet
<androbin> .queue todo replace dolor, rolod
<phenny> [androbin:todo] - sit, rolod, amet
<androbin> .queue doto replace 1, dolor
<phenny> [androbin:doto] - lorem, dolor
<androbin> .queue todo remove rolod
<phenny> [androbin:todo] - sit, amet
<androbin> .queue doto pop
<phenny> [androbin:doto] - dolor
<androbin> .queue todo random
<phenny> androbin: 'amet' is the lucky one.
<androbin> .queue todo reassign devil
<phenny> [devil:todo] - sit, amet
<androbin> .queue todo rename nope
<phenny> [devil:nope] - sit, amet
```

**Comment by Androbin** [comment]
I will also go ahead and build some unit tests from this transcript.

**Comment by sushain97** [comment]
> <androbin> .queue todo rename nope
> <phenny> [devil:nope] - sit, amet

Are you supposed to be able to rename `devil`'s queue?

**Comment by Androbin** [comment]
Yes, I'm admin ;-)

**Comment by Androbin** [comment]
I am testing both cases in the unit tests.

**Comment by Androbin** [comment]
Any more objections?

File: tools.py [hunk 1]
```diff
@@ -22,6 +22,9 @@
 logger = logging.getLogger('phenny')
 
 
+# e.g. make read/write from disc no-ops
+debugging = False
```
Comment (sushain97) [inline tools.py]: Let's do `debug = False`, it's more conventional.

File: modules/test/test_queue.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
+import unittest
+from mock import MagicMock
+from modules import queue
```
Comment (sushain97) [inline modules/test/test_queue.py]: split into internal/external please

File: modules/test/test_queue.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+import tools
+
+tools.debugging = True
+
+class TestQueue(unittest.TestCase):
+
```
Comment (sushain97) [inline modules/test/test_queue.py]: no need for this extra line

File: modules/test/test_queue.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    def expect_msg(self, message):
+        self.phenny.msg.assert_called_once_with(self.input.sender, message)
+
+    def expect_reply(self, message):
+        self.phenny.reply.assert_called_once_with(message)
+
+    def _test_mutating(self, output, revert=True):
+        input = self.input.group
+
+        self.input.group = lambda x: [None, 'todo', 'reassign', 'devil'][x]
```
Comment (sushain97) [inline modules/test/test_queue.py]: This is really hurting me. How about `self.set_input('todo', 'reassign', 'devil')`?

Comment (Androbin) [reply modules/test/test_queue.py]: Any ideas on how to refactor `test_apy.py` ?

File: tools.py [hunk 5]
```diff
@@ -22,6 +22,9 @@
 logger = logging.getLogger('phenny')
 
 
+# e.g. make read/write from disc no-ops
```
Comment (sushain97) [inline tools.py]: disk, not disc. LGTM otherwise.

**Pull request merged**